content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md

```diff
@@ -0,0 +1,331 @@
+
+You can check the service *Overview* and *Endpoint* pages per service, e.g., *frontend*:
+
+![](/img/sw-app-overview.png)
+
+![](/img/sw-app-endpoint.png)
+
+You can also check all the services from the *Data Plane* tab under  *Service Mesh* that are observed through _Envoy Metrics Service_ setup:
+
+![](/img/sw-app-svc-dataplane.png)
+
+### Online Boutique application alerting
+
+
+
+Apache SkyWalking provides an alerting mechanism to measure application performance according to a list of pre-defined metrics, e.g., *service\_resp\_time*, *service\_instance\_resp\_time*, and *service\_sla*. It will trigger alerting when some metrics reach its pre-defined thresholds. 
+
+A﻿pache SkyWalking configures the alerting using a collection of alerting rules located in */skywalking/config/alarm-settings.yml* from the SkyWalking OAP pod. You can check the content by running the following command:
+
+
+
+```markdown
+$ kukectl exec pod/skywalking-skywalking-helm-oap-bfb57fbf8-5g7k7 -n skywalking -it -- cat /skywalking/config/alarm-settings.yml 
+```
+You can define new metrics by adding new entry to the file using SkyWalking observability analysis language (OAL), or customize the existing metrics with new thresholds. 
+
+
+
+B﻿elow is the alarms page from SkyWalking UI showing all the triggered alerts for deployed Online Boutique application: 
+
+
+
+![](/img/sw-alert.png)
+
+
+
+The alarms page shows an alert * Response time of service instance frontend-549fd9954f-lvnsv of frontend is more than 1000ms in 2 minutes of last 10 minutes *. 
+
+
+
+This alert is triggered by the following metric alerting rule for the metric *service\_instance\_resp\_time*: 
+
+
+
+```markdown
+  service_instance_resp_time_rule:
+    metrics-name: service_instance_resp_time
+    op: ">"
+    threshold: 1000
+    period: 10
+    count: 2
+    silence-period: 5
+message: Response time of service instance {name} is more than 1000ms in 2 minutes of last 10 minutes
+```
+
+
+It indicates issue from the _frontend_ service instance in Online Boutique application. You can check further the service trace page to figure out the root cause for this issue.
+
+## Conclusion
+
+
+
+This blog post discussed the challenges in hybrid cloud monitoring and described the process of setting up Apache SkyWalking as a _self-hosted_ APM tool for application performance monitoring across hybrid cloud environment. Instead of taking the manual instrumentation mechanism to rebuild the application with various agents to collect and send the application metrics, we used an auto instrumentation approach with service mesh in the setup. This Envoy Access Log Service (ALS) based approach did not require any change to the deployed applications and the setup process showed that it was very easy to configure it for hybrid cloud monitoring. This integration solution can be extremely important for monitoring and visualizing application that consists of many microservices running across hybrid cloud environments.

```

Comment:
For the end of the first sentence, either add the word "a" prior to "hybrid cloud environment." OR change the word "environment" to its plural: "... across hybrid cloud environments."

For the last sentence, ad the word "an" prior to "application that consists of..."